tests/unit/acquisition/test_function.py

```diff
@@ -695,6 +698,212 @@ def test_expected_hypervolume_improvement(
+    model = QuadraticMeanAndRBFKernel()
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        BatchMonteCarloExpectedHypervolumeImprovement().prepare_acquisition_function(dataset, model)
+
+
+@pytest.mark.parametrize("sample_size", [-2, 0])
+def test_batch_monte_carlo_expected_hypervolume_improvement_raises_for_invalid_sample_size(
+    sample_size: int,
+) -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        BatchMonteCarloExpectedHypervolumeImprovement(sample_size)
+
+
+def test_batch_monte_carlo_expected_hypervolume_improvement_raises_for_invalid_jitter() -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        BatchMonteCarloExpectedHypervolumeImprovement(100, jitter=-1.0)
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "input_dim, num_samples_per_point, training_input, obj_num, variance_scale",
+    [
+        pytest.param(
+            1,
+            50_000,
+            tf.constant([[0.3], [0.22], [0.1], [0.35]]),
+            2,
+            1.0,
+            id="1d_input_2obj_model_var_1_q_1",
+        ),
+        pytest.param(
+            1,
+            50_000,
+            tf.constant([[0.3], [0.22], [0.1], [0.35]]),
+            2,
+            2.0,
+            id="1d_input_2obj_model_var_2_q_1",
+        ),
+        pytest.param(
+            2,
+            50_000,
+            tf.constant([[0.0, 0.0], [0.2, 0.5]]),
+            2,
+            1.0,
+            id="2d_input_2obj_model_var_1_q_1",
+        ),
+        pytest.param(
+            3,
+            25_000,
+            tf.constant([[0.0, 0.0, 0.2], [-0.2, 0.5, -0.1], [0.2, -0.5, 0.2]]),
+            3,
+            1.0,
+            id="3d_input_3obj_model_var_1_q_1",
+        ),
+    ],
+)
+def test_batch_monte_carlo_expected_hypervolume_improvement_can_reproduce_ehvi(
+    input_dim: int,
+    num_samples_per_point: int,
+    training_input: tf.Tensor,
+    obj_num: int,
+    variance_scale: float,
+) -> None:
+    data_num_seg_per_dim = 10  # test data number per input dim
+
+    model = _mo_test_model(obj_num, *[variance_scale] * obj_num)
+
+    mean, _ = model.predict(training_input)  # gen prepare Pareto
+    _model_based_tr_dataset = Dataset(training_input, mean)
+
+    _model_based_pareto = Pareto(mean)
+    _reference_pt = get_reference_point(_model_based_pareto.front)
+
+    qehvi_builder = BatchMonteCarloExpectedHypervolumeImprovement(sample_size=num_samples_per_point)
+    qehvi_acq = qehvi_builder.prepare_acquisition_function(_model_based_tr_dataset, model)
+    ehvi_acq = expected_hv_improvement(model, _model_based_pareto, _reference_pt)
+
+    test_xs = tf.convert_to_tensor(
+        list(itertools.product(*[list(tf.linspace(-1, 1, data_num_seg_per_dim))] * input_dim)),
+        dtype=training_input.dtype,
+    )  # [test_num, input_dim]
+    test_xs = tf.expand_dims(test_xs, -2)  # add Batch dim: q=1
+
+    npt.assert_allclose(ehvi_acq(test_xs), qehvi_acq(test_xs), rtol=1e-2, atol=1e-2)
```

Comment:
Why such low high tolerances?

Reply:
you mean `rtol` and `atol` looks quite big?

Reply:
yep

Reply:
This comes from the original difficulty of doing multi-dimensional mc integration. If we only set an `rtol=0.01`  like [`MC-EI`](https://github.com/secondmind-labs/trieste/blob/28243c5fd3ff3bec89ad3621fb2a5a2c0b34c6bd/tests/unit/acquisition/test_function.py#L738), we need an extremely large MC sample number (e.g., `1_000_000` for 2 obj case), which is slow and may cause OOM for 3d test.

Am also thinking perhaps we could later leverage Loka's customer sampler PR to reduce the variance of MC for instance in `BatchReparametrizationSampler`.

Reply:
Good idea!